base/indices.jl

```diff
@@ -19,25 +19,49 @@ abstract type IndexStyle end
 Subtype of [`IndexStyle`](@ref) used to describe arrays which
 are optimally indexed by one linear index.
 
-A linear indexing style uses one integer to describe the position in the array
+A linear indexing style uses one integer index to describe the position in the array
 (even if it's a multidimensional array) and column-major
-ordering is used to access the elements. For example,
-if `A` were a `(2, 3)` custom matrix type with linear indexing,
-and we referenced `A[5]` (using linear style), this would
-be equivalent to referencing `A[1, 3]` (since `2*1 + 3 = 5`).
+ordering is used to efficiently access the elements. This means that
+requesting [`eachindex`](@ref) from an array that is `IndexLinear` will return
+a simple one-dimensional range, even if it is multidimensional.
+
+A custom array that reports its `IndexStyle` as `IndexLinear` only needs
+to implement indexing (and indexed assingment) with a single `Int` index;
```

Comment:
With a single `Integer` instead of Int?

Reply:
The abstract fallbacks will convert `Integer`s to `Int` for you, too, if you let them.